src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertyBuilder.java

```diff
@@ -593,13 +592,19 @@ protected int _getterPriority(AnnotatedMethod m)
     protected int _setterPriority(AnnotatedMethod m)
     {
         final String name = m.getName();
-        if (name.startsWith("set") && name.length() > 3) {
-            // should we check capitalization?
+        if (validateAccessorPrefix(name, "set")) {
             return 1;
         }
         return 2;
     }
 
+    private boolean validateAccessorPrefix(String name, String prefix)
+    {
+        final int prefixLength = prefix.length();
+        return name.startsWith(prefix) && name.length() > prefixLength
+            && (!_config.isEnabled(MapperFeature.REQUIRE_CAPITALIZED_PROPERTY_ACCESSOR_NAME) || Character.isUpperCase(name.charAt(prefixLength)));
```

Comment:
One question here would be whether upper-case is required or lower-case letter not allowed -- difference being between "island" and "is_land()". I realize neither is Beans naming compatible, but I suspect latter might also be in use.